### PR TITLE
Update System.Formats.Nrbf ref sources and restore package description

### PR DESCRIPTION
**Pull request by jeffhandley**
In [Mark the System.Formats.Nrbf assembly as [Experimental] with SYSLIB5005 (#107905)](https://github.com/dotnet/runtime/pull/107905), the System.Formats.Nrbf package description was removed entirely rather than being shortened. This restores the short package description that is used in tooling when the full package README isn't displayed.

I also found the ref source for System.Formats.Nrbf and System.Reflection.Metadata was out of sync. It looks like the files were hand-edited and there were a few details missing along with inconsistencies compared to what `dotnet msbuild /t:GenerateReferenceAssemblySource` generates.

The intention is to make these corrections for .NET 9, backporting to `release/9.0` after merged to main.

### CODE REVIEW

File: src/libraries/System.Formats.Nrbf/src/System.Formats.Nrbf.csproj [hunk 1]
```diff
@@ -5,12 +5,13 @@
     <EnableDefaultItems>true</EnableDefaultItems>
     <CLSCompliant>false</CLSCompliant>
     <IsPackable>true</IsPackable>
+    <PackageDescription>Provides a safe reader for .NET Remoting Binary Format (NRBF) payloads.</PackageDescription>
+    <EnableDefaultPackageReadmeFile>true</EnableDefaultPackageReadmeFile>
```
Comment (buyaa-n) [inline src/libraries/System.Formats.Nrbf/src/System.Formats.Nrbf.csproj:9]: Not sure if this needed, but it's being removed when full package description added for other libraires, for example https://github.com/dotnet/runtime/pull/106785/files. I see the full package description added with https://github.com/dotnet/runtime/pull/106344/files but `EnableDefaultPackageReadmeFile` is not removed

**Review by buyaa-n — APPROVED** [review]
Left a NIT, overall LGTM

**Review by ericstj — APPROVED** [review]
LGTM - keeping in sync with the output of GenAPI will make this easier to maintain and ensure you don't miss things that APICompat doesn't yet catch.

**Pull request merged**